lib/ecto_ldap/adapter.ex

```diff
@@ -23,8 +24,8 @@ defmodule Ecto.Ldap.Adapter do
   # Client API
   #
   ####
-  def search(search_options) do
-    GenServer.call(__MODULE__, {:search, search_options})
+  def search(search_options, search_filter) do
+    GenServer.call(__MODULE__, {:search, search_options, search_filter}, @server_timeout)
```

Comment:
I improved the name to `search_filter` to describe the data being passed to each downstream method.